src/ralph/backends/data/async_es.py

```diff
@@ -0,0 +1,331 @@
+            new (bool): Ignored.
+
+        Yields:
+            str: The next index, data stream or alias name. (If `details` is False).
+            dict: The next index, data stream or alias details. (If `details` is True).
+
+        Raises:
+            BackendException: If a failure during indices retrieval occurs.
+        """
+        target = target if target else "*"
+        try:
+            indices = await self.client.indices.get(index=target)
+        except (ApiError, ESConnectionError) as error:
+            msg = "Failed to read indices: %s"
+            logger.error(msg, error)
+            raise BackendException(msg % error) from error
+
+        if new:
+            logger.warning("The `new` argument is ignored")
+
+        if details:
+            for index, value in indices.items():
+                yield {index: value}
+
+            return
+
+        for index in indices:
+            yield index
+
+    @enforce_query_checks
+    async def read(
+        self,
+        *,
+        query: Union[str, ESQuery] = None,
+        target: str = None,
+        chunk_size: Union[None, int] = None,
+        raw_output: bool = False,
+        ignore_errors: bool = False,
+    ) -> Iterator[Union[bytes, dict]]:
+        """Read documents matching the query in the target index and yield them.
+
+        Args:
+            query: (str or ESQuery): A query in the Lucene query string syntax or a
+                dictionary defining a search definition using the Elasticsearch Query
+                DSL. The Lucene query overrides the query DSL if present.
+                See Elasticsearch search reference for Lucene query syntax:
+                https://www.elastic.co/guide/en/elasticsearch/reference/8.6/search-search.html#search-api-query-params-q
+                See Elasticsearch search reference for query DSL syntax:
+                https://www.elastic.co/guide/en/elasticsearch/reference/8.6/search-search.html#request-body-search-query
+            target (str or None): The target Elasticsearch index name to query.
+                If target is `None`, the `default_index` is used instead.
+            chunk_size (int or None): The chunk size for reading batches of documents.
+                If chunk_size is `None` it defaults to `DEFAULT_CHUNK_SIZE`.
+            raw_output (bool): Controls whether to yield dictionaries or bytes.
+            ignore_errors (bool): Ignored.
+
+        Yields:
+            bytes: The next raw document if `raw_output` is True.
+            dict: The next JSON parsed document if `raw_output` is False.
+
+        Raises:
+            BackendException: If a failure occurs during Elasticsearch connection.
+        """  # pylint: disable=line-too-long
+        target = target if target else self.default_index
+        chunk_size = chunk_size if chunk_size else self.settings.DEFAULT_CHUNK_SIZE
+        if ignore_errors:
+            logger.warning("The `ignore_errors` argument is ignored")
```

Comment:
A `NotImplementedError` on a parameter would break the method which we do not want. As in all cases, the errors will be raised, this step should just warn the user that this parameter won't be taken into account, and the reading step should proceed

Reply:
Concerning the corrupt data, in the context of the `read` method, there is no risk, as the data stored in the backend is already `controlled`.
Concerning the second point, for now I don't know the best solution, I think that the information should be given to the user at a point, but as it's the read method, it should not raise an error